routers/legacy/router.go

```diff
@@ -59,7 +59,14 @@ type Router struct {
 // If the given OpenAPIv3 document has servers, router will use them.
 // All operations of the document will be added to the router.
 func NewRouter(doc *openapi3.T) (routers.Router, error) {
-	if err := doc.Validate(context.Background()); err != nil {
+	return NewRouterWithValidationOptions(doc)
+}
+
+// NewRouterWithValidationOptions creates a new router with specific validation options before creation.
+//
+// Does the same thing as NewRouter, but when calling `doc.Validate()`, the validation options `opts` will be provided.
+func NewRouterWithValidationOptions(doc *openapi3.T, opts ...openapi3.ValidationOption) (routers.Router, error) {
```

Comment:
Good catch! Do feel free to overload today's implementation of `NewRouter` instead of introducing a new function.

Also do the same with `gorillamux`'s router please.

Reply:
Hey, `gorillamux`'s  mentions "Assumes spec is .Validate()d" and indeed, `doc.Validate()` is not called.

What do you propose?

Note: I first tried to only change `func NewRouter` of `legacy`, but one test assumed both `legacy` and `gorillamux` to use the same signature.

Reply:
Ah well my bad sorry. Forget about `gorillamux`'s then.

Yes, just modify that test then. You'll just need to wrap `gorillamux.NewRouter` inside a func.

Reply:
Done!